ast.c

```diff
@@ -724,6 +720,50 @@ static void ast_to_zval(zval *zv, zend_ast *ast, ast_state_info_t *state) {
 			ast->kind = ZEND_AST_ASSIGN_OP;
 			ast->attr = AST_BINARY_COALESCE;
 			break;
+		case ZEND_AST_CLASS_NAME:
+			if (state->version < 70) {
+				zval name_zval;
+				ast_to_zval(&name_zval, ast->child[0], state);
+				zval class_name_zval;
+				ast_create_virtual_node_ex(
+						&class_name_zval, AST_NAME, ast->child[0]->attr, zend_ast_get_lineno(ast), state, 1, &name_zval);
+				zval const_zval;
+				ZVAL_STR(&const_zval, AST_STR(str_class));
```

Comment:
ZVAL_STR_COPY, the string may be non-interned in zts builds of older PHP versions.